catalog/core/catalog-core-solrcommands/src/main/java/org/codice/ddf/commands/solr/RestoreCommand.java

```diff
@@ -0,0 +1,319 @@
+  }
+
+  private boolean restore(
+      SolrClient client, String collection, String backupLocation, String backupName)
+      throws IOException, InterruptedException, SolrServerException {
+    if (!canRestore(client, collection)) {
+      LOGGER.warn("Unable to restore collection {}", collection);
+      return false;
+    }
+
+    CollectionAdminRequest.Restore restore =
+        CollectionAdminRequest.AsyncCollectionAdminRequest.restoreCollection(collection, backupName)
+            .setLocation(backupLocation);
+
+    String syncReqId = restore.processAsync(client);
+
+    boolean restoreComplete = false;
+
+    while (true) {
+      CollectionAdminRequest.RequestStatusResponse requestStatusResponse =
+          CollectionAdminRequest.requestStatus(syncReqId).process(client);
+      RequestStatusState requestStatus = requestStatusResponse.getRequestStatus();
+      if (requestStatus == RequestStatusState.COMPLETED) {
+        LOGGER.debug("Restore status: {}", requestStatus);
+        restoreComplete = true;
+        break;
+      } else if (requestStatus == RequestStatusState.FAILED
+          || requestStatus == RequestStatusState.NOT_FOUND) {
+        LOGGER.debug("Restore status: {}", requestStatus);
+        printErrorMessage("Restore failed. ");
+        printResponseErrorMessages(requestStatusResponse);
+        break;
+      }
+      TimeUnit.SECONDS.sleep(1);
+    }
+
+    return restoreComplete;
+  }
+
+  private String restoreAsync(
+      SolrClient client, String collection, String backupLocation, String backupName)
+      throws IOException, SolrServerException {
+
+    if (canRestore(client, collection)) {
+      CollectionAdminRequest.Restore restore =
+          CollectionAdminRequest.AsyncCollectionAdminRequest.restoreCollection(
+                  collection, backupName)
+              .setLocation(backupLocation);
+
+      String requestId = restore.processAsync(client);
+      LOGGER.debug("Restore request Id: {}", requestId);
+      return requestId;
+    } else {
+      printErrorMessage(
+          String.format("Unable to restore %s, collection already exists.", collection));
+      LOGGER.debug("Unable to restore: {}, collection already exists", collection);
+      return null;
+    }
+  }
+
+  private void performSolrCloudRestore(SolrClient client) {
+    LOGGER.debug("Restoring collection {} from {} / {}.", coreName, backupLocation, backupName);
+    printInfoMessage(
+        String.format(
+            "Restoring collection [%s] from [%s] / [%s].", coreName, backupLocation, backupName));
+    try {
+      if (asyncRestore) {
+        String requestId = restoreAsync(client, coreName, backupLocation, backupName);
+        printInfoMessage("Restore request Id: " + requestId);
+      } else {
+        boolean isSuccess = restore(client, coreName, backupLocation, backupName);
+        if (isSuccess) {
+          printInfoMessage("Restore complete.");
+        }
+      }
+    } catch (InterruptedException ie) {
```

Comment:
✏️ I would also add `InterruptedIOException` in case the I/O that occurs also handles interrupted exception.